From fd89a9e229ae5ebe959459fe62a6304c0cce0cf4 Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <ignatenkobrain@fedoraproject.org>
Date: Sat, 11 Feb 2017 08:36:24 +0100
Subject: [PATCH] add missing #include <functional>

With GCC 7.0, functional is not included transitively and we get:
In file included from qgpgmedeletejob.h:39:0,
                 from qgpgmedeletejob.cpp:38:
threadedjobmixin.h:98:33: error: 'function' in namespace 'std' does not name a template type
     void setFunction(const std::function<T_result()> &function)
                                 ^~~~~~~~

std::{function,bind,placeholders,mem_fn} are defined in functional.

References: https://bugzilla.redhat.com/show_bug.cgi?id=1417383
Signed-off-by: Igor Gnatenko <ignatenkobrain@fedoraproject.org>
---
 lang/qt/src/qgpgmenewcryptoconfig.cpp | 1 +
 lang/qt/src/threadedjobmixin.h        | 1 +
 2 files changed, 2 insertions(+)

diff --git a/lang/qt/src/qgpgmenewcryptoconfig.cpp b/lang/qt/src/qgpgmenewcryptoconfig.cpp
index 4738a03b..6901eef1 100644
--- a/lang/qt/src/qgpgmenewcryptoconfig.cpp
+++ b/lang/qt/src/qgpgmenewcryptoconfig.cpp
@@ -49,6 +49,7 @@
 #include <sstream>
 #include <string>
 #include <cassert>
+#include <functional>
 
 using namespace QGpgME;
 using namespace GpgME;
diff --git a/lang/qt/src/threadedjobmixin.h b/lang/qt/src/threadedjobmixin.h
index 32b23db9..5ad2737f 100644
--- a/lang/qt/src/threadedjobmixin.h
+++ b/lang/qt/src/threadedjobmixin.h
@@ -51,6 +51,7 @@
 #include "job.h"
 
 #include <cassert>
+#include <functional>
 
 namespace QGpgME
 {
-- 
2.11.1

